airbyte-integrations/connectors/source-mongodb-internal-poc/src/main/java/io/airbyte/integrations/source/mongodb/internal/MongoDbStateIterator.java

```diff
@@ -84,8 +100,9 @@ public boolean hasNext() {
       if (iter.hasNext()) {
         return true;
       }
-    } catch (MongoException e) {
+    } catch (final MongoException e) {
       // If hasNext throws an exception, log it and then treat it as if hasNext returned false.
+      iterThrewException = true;
```

Comment:
Am still not sure why are we swallowing the exception and not failing the sync?

Reply:
@colesnodgrass Could you please explain our rationale behind this approach?